inbox/password-storage.xml

```diff
@@ -0,0 +1,296 @@
+  </p>
+  <ul>
+    <li>SCRAM-SHA-256</li>
+    <li>SCRAM-SHA-256-PLUS</li>
+  </ul>
+  <p>
+    Clients SHOULD NOT invent their own mechanisms that have not been
+    standardized by the IETF, the XSF, or another reputable standards body.
+  </p>
+  <p>
+    Clients MUST NOT implement any mechanism with a usage status of "OBSOLETE",
+    "MUST NOT be used", or "LIMITED" in the &ianasasl;.
+    Similarly, any mechanism that depends on a hash function listed as "MUST
+    NOT" in &xep0414; MUST NOT be used.
+    This means that the following mechanisms which were commonly used with XMPP
+    in the past MUST NOT be supported:
+  </p>
+  <ul>
+    <li>CRAM-MD5 (&rfc2195;)</li>
+    <li>DIGEST-MD5 (&rfc6331;)</li>
+  </ul>
+</section1>
+<section1 topic='Client Best Practices' anchor='client'>
+  <section2 topic='Mechanism Pinning' anchor='pinning'>
+    <p>
+      Clients maintain a list of preferred SASL mechanisms, generally ordered by
+      perceived strength to enable strong authentication (RFC 6120 §6.3.3
+      Mechanism Preferences).
+      To prevent downgrade attacks by a malicious actor that has successfully
+      man in the middled a connection, or compromised a trusted server's
+      configuration, clients SHOULD implement "mechanism pinning".
+      That is, after the first successful authentication with a strong
+      mechanism, clients SHOULD make a record of the authentication and
+      thereafter only advertise and use mechanisms of equal or higher perceived
+      strength.
+    </p>
+    <p>
+      For reference, the following mechanisms are ordered by their perceived
+      strength from strongest to weakest with mechanisms of equal strength on
+      the same line.
+      This list is a non-normative example and does not indicate that these
+      mechanisms should or should not be supported:
+    </p>
+    <ol>
+      <li>EXTERNAL</li>
+      <li>SCRAM-SHA-1-PLUS, SCRAM-SHA-256-PLUS</li>
+      <li>SCRAM-SHA-1, SCRAM-SHA-256</li>
+      <li>PLAIN</li>
+      <li>DIGEST-MD5, CRAM-MD5</li>
+    </ol>
+  </section2>
+  <section2 topic='Storage' anchor='client-storage'>
+    <p>
+      Clients SHOULD always store authenticators in a trusted and encrypted
+      keystore such as the system keystore, or an encrypted store created
+      specifically for the clients use.
+      They SHOULD NOT store authenticators as plain text.
+    </p>
+    <p>
+      If clients know that they will only ever authenticate using a mechanism
+      such as SCRAM where the original password is not needed (for example if
+      the mechanism has been pinned) they SHOULD store the SCRAM bits or the
+      hashed and salted password instead of the original password.
+      However, if backwards compatibility with servers that only support the
+      PLAIN mechanism or other mechanisms that require using the original
+      password is required, clients MAY choose to store the original password
+      so long as an appropriate keystore is used.
```

Comment:
```suggestion
      as long as an appropriate keystore is used.
```

Reply:
I think this one is okay, otherwise you wind up with "as long as" which sounds a bit odd to me (too many "as"'s)